nodetool Refresh nemesis not able to detect "Resharded" and nemesis fails (but there was a Resharded)

### Versions
* **SCT**: master
* **scylla**: master

### Logs
* test_id: eadde21f-ad93-476f-a546-842a4fea2708
* job log: https://jenkins.scylladb.com/view/master/job/scylla-master/view/master-test/job/longevity/job/longevity-200gb-48h/123/

i see in sct.log:
```
< t:2021-06-18 09:05:00,331 f:base.py         l:140  c:RemoteCmdRunner      p:DEBUG > Command "sudo -u scylla tar xvfz /tmp/keyspace1.standard1.tar.gz -C /var/lib/scylla/data/keyspace1/standard1-3d0c3510
cfe611eb96c70f5316c4eada/upload/" finished with status 0
< t:2021-06-18 09:05:00,332 f:remote_base.py  l:520  c:RemoteCmdRunner      p:DEBUG > Running command "sudo rm -f /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/schema.c
ql"...
< t:2021-06-18 09:05:00,396 f:base.py         l:140  c:RemoteCmdRunner      p:DEBUG > Command "sudo rm -f /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/schema.cql" fini
shed with status 0
< t:2021-06-18 09:05:00,396 f:remote_base.py  l:520  c:RemoteCmdRunner      p:DEBUG > Running command "sudo rm -f /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/manifest
.json"...
< t:2021-06-18 09:05:00,429 f:base.py         l:140  c:RemoteCmdRunner      p:DEBUG > Command "sudo rm -f /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/manifest.json" f
inished with status 0
< t:2021-06-18 09:05:00,429 f:nemesis.py      l:994  c:sdcm.nemesis         p:DEBUG > sdcm.nemesis.LimitedChaosMonkey: Loading 1000 keys to longevity-200gb-48h-verify-limited--db-node-eadde21f-1 by refresh
< t:2021-06-18 09:05:00,430 f:remote_base.py  l:520  c:RemoteCmdRunner      p:DEBUG > Running command "/usr/bin/nodetool -u cassandra -pw 'cassandra'  refresh -- keyspace1 standard1"...
< t:2021-06-18 09:05:00,432 f:file_logger.py  l:80   c:sdcm.sct_events.file_logger p:INFO  > 2021-06-18 09:05:00.430: (NodetoolEvent Severity.NORMAL) period_type=not-set event_id=3b1a608f-29e6-4151-bdf4-4f42981cd1eb: type=refresh subtype=start node=longevity-200gb-48h-verify-limited--db-node-eadde21f-1
< t:2021-06-18 09:05:00,437 f:connectionpool.py l:227  c:urllib3.connectionpool p:DEBUG > Starting new HTTP connection (1): 10.0.0.168:3000
< t:2021-06-18 09:05:00,443 f:connectionpool.py l:452  c:urllib3.connectionpool p:DEBUG > http://10.0.0.168:3000 "POST /api/annotations HTTP/1.1" 200 39
< t:2021-06-18 09:05:03,379 f:connectionpool.py l:227  c:urllib3.connectionpool p:DEBUG > Starting new HTTP connection (1): 10.0.0.168:9093
< t:2021-06-18 09:05:03,380 f:connectionpool.py l:452  c:urllib3.connectionpool p:DEBUG > http://10.0.0.168:9093 "GET /api/v2/alerts?active=1 HTTP/1.1" 200 1154
< t:2021-06-18 09:05:03,382 f:connectionpool.py l:227  c:urllib3.connectionpool p:DEBUG > Starting new HTTP connection (1): 10.0.0.168:9093
< t:2021-06-18 09:05:03,384 f:connectionpool.py l:452  c:urllib3.connectionpool p:DEBUG > http://10.0.0.168:9093 "GET /api/v2/alerts HTTP/1.1" 200 1154
< t:2021-06-18 09:05:05,028 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > total,     291262576,   17850,   17850,   17850,    13.9,     9.0,    40.1,    70.5,   197.3,   317.7,15605.0,  0.00098,      0,      0,       0,       0,       0,       0
< t:2021-06-18 09:05:08,147 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > 09:05:08.148 [cluster1-timeouter-0] DEBUG c.d.driver.core.RequestHandler - onTimeout triggered but the response was completed by another thread, cancelling (retryCount = 0, queryState = QueryState(count=0, inProgress=false, cancelled=false), queryStateRef = QueryState(count=0, inProgress=false, cancelled=false))
< t:2021-06-18 09:05:10,034 f:base.py         l:220  c:RemoteCmdRunner      p:DEBUG > total,     291352104,   17906,   17906,   17906,    14.0,     9.0,    36.0,   107.9,   206.2,   312.5,15610.0,  0.00098,      0,      0,       0,       0,       0,       0
< t:2021-06-18 09:05:10,546 f:base.py         l:140  c:RemoteCmdRunner      p:DEBUG > Command "/usr/bin/nodetool -u cassandra -pw 'cassandra'  refresh -- keyspace1 standard1" finished with status 0
< t:2021-06-18 09:05:10,546 f:cluster.py      l:2611 c:sdcm.cluster_aws     p:DEBUG > Node longevity-200gb-48h-verify-limited--db-node-eadde21f-1 [13.51.156.141 | 10.0.1.212] (seed: True): Command '/usr/bin/nodetool -u cassandra -pw 'cassandra'  refresh -- keyspace1 standard1' duration -> 10.115686147000815 s
< t:2021-06-18 09:05:10,546 f:events_processes.py l:143  c:sdcm.sct_events.events_processes p:DEBUG > Get process `MainDevice' from EventsProcessesRegistry[lod_dir=/home/ubuntu/sct-results/20210618-030829-059885,id=0x7fe7b4dc40a0,default=True]
< t:2021-06-18 09:05:10,547 f:file_logger.py  l:80   c:sdcm.sct_events.file_logger p:INFO  > 2021-06-18 09:05:10.546: (NodetoolEvent Severity.NORMAL) period_type=not-set event_id=dfd60a69-725a-4669-87ac-0f14535f8f55: type=refresh subtype=end node=longevity-200gb-48h-verify-limited--db-node-eadde21f-1 duration=10.115686147000815s
< t:2021-06-18 09:05:10,548 f:nemesis.py      l:1130 c:sdcm.nemesis         p:ERROR > sdcm.nemesis.LimitedChaosMonkey: ('Exception in random_disrupt_method %s: %s', 'nodetool_refresh', AssertionError("Resharding wasn't run"))
< t:2021-06-18 09:05:10,551 f:connectionpool.py l:227  c:urllib3.connectionpool p:DEBUG > Starting new HTTP connection (1): 10.0.0.168:3000
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR > sdcm.nemesis.LimitedChaosMonkey: Unhandled exception in method <function LimitedChaosMonkey.disrupt at 0x7fe7b7c14dc0> < t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR > sdcm.nemesis.LimitedChaosMonkey: Unhandled exception in method <function LimitedChaosMonkey.disrupt at 0x7fe7b7c14dc0>
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR > Traceback (most recent call last):
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 2970, in wrapper
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR >     result = method(*args, **kwargs)
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 3379, in disrupt
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR >     self.call_random_disrupt_method(disrupt_methods=['disrupt_nodetool_cleanup', 'disrupt_nodetool_decommission',
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 1120, in call_random_disrupt_method
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR >     self.execute_disrupt_method(disrupt_method)
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 1127, in execute_disrupt_method
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR >     disrupt_method()
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 1045, in disrupt_nodetool_refresh
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR >     validate_resharding_after_refresh(resharding_done)
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR >   File "/home/ubuntu/scylla-cluster-tests/sdcm/nemesis.py", line 1022, in validate_resharding_after_refresh
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR >     assert resharding_done, "Resharding wasn't run"
< t:2021-06-18 09:05:10,549 f:nemesis.py      l:2980 c:sdcm.nemesis         p:ERROR > AssertionError: Resharding wasn't run
< t:2021-06-18 09:05:10,552 f:nemesis.py      l:2992 c:sdcm.nemesis         p:DEBUG > sdcm.nemesis.LimitedChaosMonkey: Refresh keyspace1.standard1 on longevity-200gb-48h-verify-limited--db-node-eadde21f-2 duration -> 19 s
```

but in the node-1 (where we started the to run the command `nodetool refresh`) i see:
```
2021-06-18T09:05:02+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-1 !INFO    | scylla:  [shard 2] storage_service - Loading new SSTables for keyspace=keyspace1, table=standard1, load_and_stream=false, primary_replica_only=false
2021-06-18T09:05:02+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-1 !INFO    | scylla:  [shard 2] database - Resharding 66kB for keyspace1.standard1
2021-06-18T09:05:02+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-1 !INFO    | scylla:  [shard 0] compaction - [Reshard keyspace1.standard1 445b9cb0-d014-11eb-a0bb-a41a05f08e92] Resharding [/var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/mc-1-big-Data.db:level=0:origin=, ]
2021-06-18T09:05:03+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-1 !INFO    | scylla:  [shard 0] compaction - [Reshard keyspace1.standard1 445b9cb0-d014-11eb-a0bb-a41a05f08e92] Resharded 1 sstables to [/var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-23-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-14-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-27-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-20-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-16-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-21-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-18-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-22-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-26-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-17-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-24-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-15-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-19-big-Data.db:level=0, /var/lib/scylla/data/keyspace1/standard1-3d0c3510cfe611eb96c70f5316c4eada/upload/md-25-big-Data.db:level=0, ]. 88kB to 1MB (~1540% of original) in 286ms = 4MB/s. ~1024 total partitions merged to 1000.
2021-06-18T09:05:03+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-1 !INFO    | scylla:  [shard 2] database - Resharded 66kB for keyspace1.standard1 in 0.31 seconds, 213kB/s
```

it means that, for some reason, the `node.follow_system_log()` command did not find the pattern:
```
search_reshard = node.follow_system_log(patterns=['Resharded.*\[/'])
node.run_nodetool(sub_cmd="refresh", args="-- keyspace1 standard1")
resharding_done = list(search_reshard)
return resharding_done
```

and it returned `False`, making the nemesis to fail in this point.

i looked at it with @juliayakovlev , and everything seems to be correct, but maybe @dkropachev might understand better what could be the cause for us to be missing that message in the log.

--- a/sdcm/nemesis.py
+++ b/sdcm/nemesis.py
@@ -63,6 +63,7 @@ from sdcm.db_stats import PrometheusDBStats
 from sdcm.remote.libssh2_client.exceptions import UnexpectedExit as Libssh2UnexpectedExit
 from sdcm.cluster_k8s import PodCluster, ScyllaPodCluster
 from sdcm.nemesis_publisher import NemesisElasticSearchPublisher
+from sdcm.wait import wait_for
 from test_lib.compaction import CompactionStrategy, get_compaction_strategy, get_compaction_random_additional_params
 from test_lib.cql_types import CQLTypeBuilder
 
@@ -1002,12 +1003,15 @@ class Nemesis:  # pylint: disable=too-many-instance-attributes,too-many-public-m
             # So we need to validate that resharded files are placed in the "upload" folder before moving.
             # Find the compaction output that reported about the resharding
 
-            search_reshard = node.follow_system_log(patterns=['Resharded.*\[/'])
+            system_log_follower = node.follow_system_log(patterns=['Resharded.*\[/'])
             node.run_nodetool(sub_cmd="refresh", args="-- keyspace1 standard1")
-            resharding_done = list(search_reshard)
-            return resharding_done
+            return system_log_follower
 
-        def validate_resharding_after_refresh(resharding_done):
+        @timeout(
+            timeout=60,
+            allowed_exceptions=(AssertionError,),
+            message="Waiting for resharding completion message to appear in logs")
+        def validate_resharding_after_refresh(system_log_follower):
             """
             # Validate that files after resharding were saved in the "upload" folder.
             # Example of compaction output:
@@ -1024,10 +1028,11 @@ class Nemesis:  # pylint: disable=too-many-instance-attributes,too-many-public-m
             #   /var/lib/scylla/data/keyspace1/standard1-9fbed8d0f8c211ea9bb1000000000000/upload/md-16-big-Data.db:level=0,
             #   ]. 91MB to 92MB (~100% of original) in 5009ms = 18MB/s. ~370176 total partitions merged to 370150
             """
-            assert resharding_done, "Resharding wasn't run"
-            self.log.debug(f"Found resharding: {resharding_done}")
+            resharding_logs = list(system_log_follower)
+            assert resharding_logs, "Resharding wasn't run"
+            self.log.debug(f"Found resharding: {resharding_logs}")
 
-            for line in resharding_done:
+            for line in resharding_logs:
                 # Find all files that were created after resharding
                 for one_file in re.findall(r"(/var/.*?),", line, re.IGNORECASE):
                     # The file path have to include "upload" folder
@@ -1046,8 +1051,8 @@ class Nemesis:  # pylint: disable=too-many-instance-attributes,too-many-public-m
 
             # Executing rolling refresh one by one
             for node in self.cluster.nodes:
-                resharding_done = do_refresh(node)
-                validate_resharding_after_refresh(resharding_done)
+                system_log_follower = do_refresh(node)
+                validate_resharding_after_refresh(system_log_follower)
 
             # Verify that the special key is loaded by SELECT query
             result = self.target_node.run_cqlsh(query_verify)

